openid-4-verifiable-presentations-1_0.md

```diff
@@ -1492,7 +1492,139 @@ issuers in Self-Sovereign Identity ecosystems using TRAIN</title>
         </front>
 </reference>
 
-# Usage with Various Credential Formats {#alternative_credential_formats}
+<reference anchor="w3c.digital_credentials_api" target="https://wicg.github.io/digital-credentials/">
+        <front>
+          <title>Digital Credentials API</title>
+		  <author fullname="Marcos Caceres">
+            <organization>Apple Inc.</organization>
+          </author>
+          <author fullname="Sam Goto">
+            <organization>Google</organization>
+          </author>
+        </front>
+</reference>
+
+# OpenID4VP profile for the W3C Digital Credentials API
+
+This section defines a profile of OpenID4VP for use with the W3C Digital Credentials API [@!w3c.digital_credentials_api].
+
+The W3C Digital Credentials API defines a Web Platform API which allows web sites acting as Verifiers
+to request the presentation of Verifiable Credentials. The API itself does not define a credential exchange protocol
+but can be used with multiple protocols. The user agent working together with other layers of the platform/operating system and based on the permission of the End-User will send to the Wallet of the End-User's choice the request data along with the web origin of the Verifier.
+
+The design of this OpenID4VP profile utilizes the mechanisms of the W3C Digital Credentials API while also allowing to leverage advanced security features of OpenID4VP, if needed. It also defines the OpenID4VP request parameter that MAY be used with the W3C Digital Credentials API.
+
+The Digital Credentials API offers several advantages for implementers of both Verifiers and Wallets. 
+
+Firstly, the API serves as a privacy-preserving alternative to invoking Wallets via URLs, particularly custom schemes. The user agent and app platform will only invoke a Wallet if the user confirms the request based on contextual information about the credential request and the requestor (verifier). 
+
+Secondly, the user will always be returned to the initial context, typically a browser tab, when the request has been fulfilled (or aborted), which results in an improved user experience.
+
+Thirdly, cross-device requests benefit from the use of secure transports with proximity checks, which are handled by the OS platform using FIDO CTAP 2.2 with hybrid transports.
+
+And lastly, as part of the request, the Wallet is provided with information about the Verifier's origin as authenticated by the user agent, which is important for phishing resistance.  
+
+## Protocol
+
+The value of the `protocol` parameter of the W3C Digital Credentials API MUST be set to `urn:openid.net:oid4vp` for this profile.
+
+## Request {#browser_api_request}
+
+The `request` member of the W3C Digital Credentials API [@!w3c.digital_credentials_api] MUST contain a valid OpenID4VP Authorization Request, where every OpenID4VP Authorization Request parameter is represented as a top-level JavaScript object member.
```

Comment:
Be careful when we talk about "valid", for "valid" will need to be linked to a validity checker.... that's ok... but we need to establish clearly what a "valid" thing is, and in a future/backwards compatible way.

Reply:
I think we can remove 'MUST' and 'valid' without losing anything (and in fact we gain something, as a 'valid' request via the browser API may not be a valid OpenID4VP request over the non-browser API, e.g. the unsigned browser API request below is not valid when passed via a custom scheme):

```suggestion
The `request` member of the W3C Digital Credentials API [@!w3c.digital_credentials_api] contains an OpenID4VP Authorization Request, where every OpenID4VP Authorization Request parameter is represented as a top-level JavaScript object member.
```

Reply:
Yeah, that makes sense @Sakurann. Making it just a "statement of fact" is a good suggestion. 

We will still need to deal with invalid requests somewhere, but we can get to that.

Reply:
```suggestion
The `request` member of the W3C Digital Credentials API [@!w3c.digital_credentials_api] contains a valid OpenID4VP Authorization Request, where every OpenID4VP Authorization Request parameter is represented as a top-level JavaScript object member.
```

why not `contains a valid` request. as a specification, we have to tell the developers to put a valid request. (even if it might end up being invalid for various reasons)

Reply:
Because:

1.  "valid request" is ambiguous in this context and causes potential confusion (a 'valid' request via the browser API may not be a valid OpenID4VP request over the non-browser API, e.g. the unsigned browser API request below is not valid when passed via a custom scheme)
2. it's not something specifications normally say (e.g. neither OIDC nor RFC6749 say it)

Reply:
well, in this case, it is pretty clear that valid request applies to a openid4vp part only but sure, I am not going to die on this hill